packages/@sanity/form-builder/src/inputs/StringInput.tsx

```diff
@@ -19,29 +20,60 @@ const StringInput = React.forwardRef(function StringInput(
     [markers]
   )
 
-  const handleChange = React.useCallback(
+  const [localValue, setlocalValue] = useState(null)
+  const inputRef = useForwardedRef(forwardedRef)
+
+  const throttledSubmit = useMemo(
+    () =>
+      throttle(
+        () => {
+          const inputValue = inputRef.current.value
+          onChange(PatchEvent.from(inputValue ? set(inputValue) : unset()))
+          setlocalValue(null)
+        },
+        500,
+        {leading: false, trailing: true}
+      ),
+    [inputRef, onChange]
+  )
+
+  const handleChange = useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
       const inputValue = event.currentTarget.value
-      onChange(PatchEvent.from(inputValue ? set(inputValue) : unset()))
+      setlocalValue(inputValue)
+      throttledSubmit()
     },
-    [onChange]
+    [throttledSubmit]
   )
 
+  useEffect(() => () => throttledSubmit(), [throttledSubmit])
```

Comment:
should be `throttledSubmit.flush()` here I think?

Reply:
Yes! I was actually thinking about this in the weekend. My idea was just to call onChange there directly.